packages/remix-react/routes.tsx

```diff
@@ -177,23 +179,59 @@ function createDataFunction(
   routeModules: RouteModules,
   isAction: boolean
 ): LoaderFunction | ActionFunction {
-  return async ({ request }) => {
+  let doFetch = async (request: Request) => {
+    if (isAction && !route.hasAction) {
+      let msg =
+        `Route "${route.id}" does not have an action, but you are trying ` +
+        `to submit to it. To fix this, please add an \`action\` function to the route`;
+      console.error(msg);
+      throw new Error(msg);
+    } else if (!isAction && !route.hasLoader) {
+      return null;
+    }
+
+    let result = await fetchData(request, route.id);
```

Comment:
I wonder if `doFetch` could stop here so we get all the same error/redirect/catch handling whether they hit the server or not?